File: server-lite/src/model/event.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, Boolean
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from sqlalchemy.dialects.postgresql import UUID
from database import Base

class Event(Base):
    __tablename__ = "events"
    
    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(200), nullable=False)
    description = Column(Text, nullable=False)
    location = Column(String(200))
    is_remote = Column(Boolean, default=False)
    date = Column(DateTime(timezone=True), nullable=False)
    is_active = Column(Boolean, default=True)
    
    employer_id = Column(UUID, ForeignKey("users.id"), nullable=False)
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    
    employer = relationship("EmployerProfile", back_populates="events")

File: app/EmploymentApp/Models/Event.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EmploymentApp.Models
{
    public class Event
    {
        public int Id { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public string Location { get; set; }

        public bool IsRemote { get; set; }

        public DateTime Date { get; set; }

        public bool IsActive { get; set; }

        public int EmployerId { get; set; }

        public string FormattedDate =>
            Date.ToString("dd MMMM yyyy, HH:mm", new System.Globalization.CultureInfo("ru-RU"));

        public string EventLocation =>
            IsRemote ? "Онлайн" : Location;

        public Event(int id, string title, string description, string location, bool isRemote,
            DateTime date, int employerId, bool isActive = true )
        {
            Id = id;
            Title = title;
            Description = description;
            Location = location;
            IsRemote = isRemote;
            Date = date;
            IsActive = isActive;
            EmployerId = employerId;
        }
    }
}

